Custom Logo(s) via Image-Upload in Panel Settings

bashgeek opened the pull request:
Again a rather simple change, but possibly good quality of life update for some of us.

This will add the ability to place a custom logo file (logo_custom_login.png) right along with the already existing custom logo (logo_custom.png). If this new file exists it'll be used as the logo shown during Login and Password Resets and the "old" logo_custom will only be used in the header in the dashboard after, if it doesn't exist it'll use the logo_custom.png for both, if both don't exist, the default logo is shown - as it has before.

You may ask why? The logo shown during Login is way bigger and gives different possibilities in terms of available vertical height to show a tag line or name together with the logo in the image. If you do that now, it looks great during login but very terrible in the dashboard, because it shows the same logo but with a fixed height of 30 pixels - it would be better to just show a simple logo without any text. You could probably achieve the same thing by introducing a custom theme, but that seems like a maintenance overkill for a simple thing like this and the default theme is more than fine.

It doesn't have any breaking changes, either you add the file or you don't. If you don't do anything it'll simply works as before.

Edit: Corrected the path for both images in the .gitignore file

d00p commented:
Good idea, sadly this option is pretty unknown because it is not a setting. Maybe it would be a good idea to have these two logo's uploadable via the settings? What do you think?

bashgeek commented:
@d00p Sure, why not. I looked a bit into it and couldn't fine even one file-upload in all the settings, so it's definitely a bigger change. That'll be also quite painful to integrate into the Settings Export/Import functionality. How about instead of file-uploads we make it simple text inputs and ask for a URL for those images? That would be more or less in line with other existing and similar settings and is easily imported/exported.

d00p commented:
That's right, there is no file-upload anywhere, URL hm...if external, will be blocked by our headers (CORS).
Settings import/export could easily be done with base64-ing the file but would require additional file-i/o logich.

Maybe we just leave it with the file-names for now ;)

bashgeek commented:
Might as well do it properly now :D

Removed the whole custom file logic and introduced an "image" setting field in the recent commit, where an admin can upload those images in the panel settings. Works fine now, just missing the import/export logic for now - will add that later.

@d00p Looks good so far?

bashgeek commented:
Alright, that was faster than expected: Import/Export now also includes the images and restores them.

bashgeek commented on .gitignore:
```diff
@@ -12,9 +12,9 @@ logs/*
 .well-known
 .idea
 *.iml
+img/
```
What do you mean exactly? Can't even see this folder in the repo, so it shouldn't exist anywhere.

d00p replied:
in the templates `templates/Sparkle/assets/img`

And if you did not mean to block this one, then why add `img/` in there after all?

bashgeek replied:
This is the folder in the root where the new images handled by the new image field type will be saved - they won't go to the theme folder. Just to make sure I'm not committing my custom uploaded images to git ;-)

d00p replied:
Got it, new img/ folder for the custom logos in root directory. Then it makes totally sense :)

bashgeek commented on lib/Froxlor/Settings/Store.php:
```diff
@@ -367,4 +367,62 @@ public static function storeSettingWebserverFcgidFpmUser($fieldname, $fielddata,
+            $path = \Froxlor\Froxlor::getInstallDir().'/img/';
+
+            // New file?
+            if ($_FILES[$fieldname]['tmp_name']) {
```
Added a proper mime-check now, however, it was already pretty hard to do so since the upload file selector was limited to jpg/png already and it requires an admin-login in the panel in the first place.

The naming of those files is already static and set in the field settings, the user cannot change that.

d00p replied:
I know...I just remembered the guy reporting that "possible security issue" when installing froxlor and having the userdata file in /tmp etc.etc.etc. totally overkill but ...still a potential way so, better safe than sorry

bashgeek replied:
Yeah, added the mime-type check in the last commit.

bashgeek commented on install/updates/froxlor/0.10/update_0.10.inc.php:
```diff
@@ -825,3 +825,13 @@
     \Froxlor\Froxlor::updateToDbVersion('202106270');
 }
+
+if (\Froxlor\Froxlor::isDatabaseVersion('202107070')) {
```
Makes sense, done! :-)
